src/grt/src/MakeWireParasitics.cpp

```diff
@@ -114,54 +128,155 @@ sta::Pin* MakeWireParasitics::staPin(Pin& pin)
     return network_->dbToSta(pin.getITerm());
 }
 
-void MakeWireParasitics::makeRouteParasitics(odb::dbNet* net,
-                                             std::vector<GSegment>& routes)
+void MakeWireParasitics::makeWireParasitics(odb::dbNet* net,
+                        std::vector<GSegment>& routes,
+                        int& routes_cnt)
 {
+  sta::Units* units = sta_->units();
   for (GSegment& route : routes) {
-    sta::ParasiticNode* n1
-        = ensureParasiticNode(route.init_x, route.init_y, route.init_layer);
-    sta::ParasiticNode* n2
-        = ensureParasiticNode(route.final_x, route.final_y, route.final_layer);
     int wire_length_dbu = route.length();
-    sta::Units* units = sta_->units();
-    float res = 0.0;
-    float cap = 0.0;
-    if (wire_length_dbu == 0) {
-      // via
-      int lower_layer = min(route.init_layer, route.final_layer);
-      odb::dbTechLayer* cut_layer
-          = tech_->findRoutingLayer(lower_layer)->getUpperLayer();
-      res = cut_layer->getResistance();  // assumes single cut
-      cap = 0.0;
-      debugPrint(logger_,
-                 GRT,
-                 "est_rc",
-                 1,
-                 "{} -> {} via r={}",
-                 parasitics_->name(n1),
-                 parasitics_->name(n2),
-                 units->resistanceUnit()->asString(res));
-    } else if (route.init_layer == route.final_layer) {
+    if (wire_length_dbu > 0 &&
+        route.init_layer == route.final_layer) {
```

Comment:
Can you have wire_length_dbu > 0 && route.init_layer **!=** route.final_layer ?  makeViaParasitics only uses

if (wire_length_dbu == 0) {

Reply:
No, it cannot happen. I'll remove the unnecessary comparison.